src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -25,16 +25,29 @@
 #ifndef SHARE_NMT_VIRTUALMEMORYTRACKER_HPP
 #define SHARE_NMT_VIRTUALMEMORYTRACKER_HPP
 
-#include "memory/allocation.hpp"
-#include "memory/metaspace.hpp" // For MetadataType
-#include "memory/metaspaceStats.hpp"
 #include "nmt/allocationSite.hpp"
-#include "nmt/nmtCommon.hpp"
+#include "nmt/vmatree.hpp"
+#include "nmt/regionsTree.hpp"
 #include "runtime/atomic.hpp"
-#include "utilities/linkedlist.hpp"
 #include "utilities/nativeCallStack.hpp"
 #include "utilities/ostream.hpp"
 
+// VirtualMemoryTracker (VMT) is the internal class of NMT that only the MemTracker class uses it for performing the NMT operations.
+// All the Hotspot code use only the MemTracker interface to register the memory operations in NMT.
+// Memory regions can be reserved/committed/uncommitted/released by calling MemTracker API which in turn call the corresponding functions in VMT.
+// VMT uses RegionsTree to hold and manage the memory regions. Each region has two nodes that each one has address of the region (start/end) and
+// state (reserved/released/committed) and MemTag of the regions before and after it.
+//
+// The memory operations of Reserve/Commit/Uncommit/Release (RCUR) are tracked by updating/inserting/deleting the nodes in the tree. When an operation
+// changes nodes in the tree, the summary of the changes is returned back in a SummaryDiff struct. This struct shows that how much reserve/commit amount
+// of any specific MemTag is changed. The summary of every operation is accumulated in VirtualMemorySummary class.
+//
+// Not all operations are valid in VMT. The following predicates are checked before the operation is applied to the tree and/or VirtualMemorySummary:
+//   - committed size of a MemTag should be <= of its reserved size
+//   - uncommitted size of a MemTag should be <= of its committed size
+//   - released size of a MemTag should be <= of its reserved size
+
```

Comment:
I don't believe that these are checked, right? So this can be deleted.

Reply:
As said at the end of line, when they are applied to the VirtualMemorySummary they will be checked.
Is it OK if I limit/wrap the comments to 80-columns text?

Reply:
Aha, that's when they're checked. Feel free to do so if you want, but I don't see any problem with going above 80 columns.